sdk/deviceupdate/Azure.Iot.DeviceUpdate/tests/ServiceClientTestEnvironment.cs

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.TestFramework;
+
+namespace Azure.Iot.DeviceUpdate.Tests
+{
+    /// <summary>
+    /// A service client test environment.
+    /// </summary>
+    /// <seealso cref="TestEnvironment"/>
+    public class ServiceClientTestEnvironment : TestEnvironment
+    {
+        /// <summary>
+        /// Initializes a new instance of the ServiceClientTestEnvironment class.
+        /// </summary>
+        public ServiceClientTestEnvironment()
+        { }
+
+        /// <summary>
+        /// ADU account identifier
+        /// </summary>
+        public string AccountEndpoint => GetRecordedVariable("DEVICEUPDATE_ACCOUNT_ENDPOINT");
```

Comment:
I haven't noticed this resource being created in test-resources.json

Reply:
Same for other variables. To be able to run live tests all resources required need to be created after deploying test-resources.json

Reply:
That's right. The tests are written in such a way, that I need to set certain env. variables, run sample and it will populate things. After that I can run the test suite and it will collect test sessions.

You should be able to replay test sessions since the test sessions will capture environment variables.

Reply:
It's fine for the initial PR but won't work with nightly CI runs, you won't be able to set additional variables there. Is there a way to create all resources via the test-resource.json without requiring manual steps?